sources/lib/structures/typed.ml

```diff
@@ -106,6 +109,7 @@ and 'a tatom =
   | TAle of ('a tterm, 'a) annoted list
   | TAlt of ('a tterm, 'a) annoted list
   | TApred of ('a tterm, 'a) annoted * bool (* true <-> negated *)
+  | TTisConstr of ('a tterm, 'a) annoted  * Hstring.t
```

Comment:
I know this exists in the syntax, and thus deserve a case in the parsed AST. In the typed AST, I think it would be better for either everything to be pattern matching, or everything to be if_then_else using the `isConstr` case, for simplicity. That way, code handling pattern matching only has one case instead of two.